src/inference/src/dev/core_impl.cpp

```diff
@@ -800,11 +798,57 @@ ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& full_device_n
     } catch (ov::Exception&) {
     }
 
+    // if device_name represents virtual device (software plugin), we need to add properties for
+    // its sub-devices, which are either virtual or hardware devices
+    if (is_virtual_device(device_name)) {
+        const auto priority_prop_name_it = options.find(priority_prop_name);
+        const std::string device_priorities_value =
+            priority_prop_name_it != options.end() ?
+                                                   // take user-set value
+                priority_prop_name_it->second.as<std::string>()
+                                                   :
+                                                   // or extract from the plugin if it's set globally
+                get_property(device_name, priority_prop_name, options).as<std::string>();
+
+        std::vector<std::string> sub_devices;
+        if (device_name == "HETERO")
+            sub_devices = ov::DeviceIDParser::get_hetero_devices(device_priorities_value);
+        else if (device_name == "AUTO" || device_name == "MULTI")
+            sub_devices = ov::DeviceIDParser::get_multi_devices(device_priorities_value);
+        else if (device_name == "BATCH")
+            sub_devices = {ov::DeviceIDParser::get_batch_device(device_priorities_value)};
+
+        /** go over flattened_config:
+         * - device::properties needs to be bypassed by as is to supported_config, because they
+         *   are actually handled / flattened in later calls
+         * - other properties need to be passed to get_supported_property call for sub devices
+         */
+        ov::AnyMap simple_properties;
+        for (const auto& property : flattened_config) {
+            const auto is_secondary_property = property.first.find(ov::device::properties.name()) != std::string::npos;
+            if (is_secondary_property) {
+                supported_config[property.first] = property.second;
+            } else {
+                simple_properties[property.first] = property.second;
+            }
+        }
+
+        // check simple / flat properties on priority devices (ov::device::priorities)
+        for (const auto& sub_device : sub_devices) {
+            auto sub_device_supported_properties = get_supported_property(sub_device, simple_properties);
+            for (auto&& kvp : sub_device_supported_properties) {
+                supported_config[kvp.first] = kvp.second;
```

Comment:
Maybe such properties should be wrapped using device properties for appropriate device? Otherwise virtual plugin can raise an exception facing unknown property

Reply:
> Otherwise virtual plugin can raise an exception facing unknown property

Why they are unknown? If they are known by specific device which returned it via `get_supported_property(sub_device, simple_properties)` call.

Could you please provide simple example ?

Reply:
> > Otherwise virtual plugin can raise an exception facing unknown property
> 
> Why they are unknown? If they are known by specific device which returned it via `get_supported_property(sub_device, simple_properties)` call.
> 
> Could you please provide simple example ?

I mean that you get properties for underlying device and put into config for virtual. Virtual plugin can raise an exception if we agreed that it should support only own properties.

Reply:
I would like to avoid supporting a superset of config inputs and silently ignoring some of them at runtime without any errors.

It may confuse user if there is any typo in config.

Reply:
We don't know beforehand which options are really required or not, because different devices can be selected in AUTO or HETERO. So, how do you propose to warn about wrong config?

Reply:
I would like to continue decoupling dependencies between plugins and removing properties in meta plugins just for transmission in initial discussion. How about the below approach:

- User only can set specific properties to specific device by using `ov::device::properties()` as user input properties. All primary properties in `compile_model()` is user input properties for input device only. 
  - For example `core.compile_model(model, "AUTO:HETERO,CPU", ov::device::priorities("HETERO:GPU,CPU", hetero_properties), ov::device::priorities("GPU", gpu_properties), primary_properties);`, primary_properties is user input properties for AUTO plugin only.
- Core pass all primary properties and secondary properties to plugin. Each plugin (included both meta plugin and HW plugin) only verify primary properties for this device and throw exception if primary properties are not supported. 
  - In your example `core.compile_model(model, "AUTO", ov::device::priorities("HETERO"), ov::device::properties("HETERO", ov::device::priorities("CPU,GPU")), ov::num::streams(5));`. AUTO will throw exception because AUTO does not support `ov::num::streams(5)`.
- Meta plugin can set properties as primary properties to next plugin during compile_model. Core will query supported properties from next plugin first and silently ignore some primary properties set by meta plugin. Then Core merge primary properties and secondary properties as one list and set to next plugin during compile_model().
  - For example `core.compile_model(model, "AUTO:XPU,CPU", ov::hint::CUMULATIVE_THROUGHPUT);`, AUTO will set tput hint to XPU like `core.compile_model(model, "XPU", ov::hint::THROUGHPUT);`. If XPU do not support ov::hint, Core will silently ignore it.
  - If user set  `core.compile_model(model, "AUTO:XPU,CPU", ov::device::properties("XPU", ov::hint::LATENCY), ov::hint::CUMULATIVE_THROUGHPUT);`, Core will merge `ov::hint::LATENCY` and set it to XPU. Then XPU throw exception since XPU do not support it.

@ilya-lavrenov @nkogteva How do you think this approach?

Reply:
@wangleis 

1. I don't think that it is user friendly. Let's imagine situation when user just want to specify peformance/execution hint or turn on peformance counters. Instead of just specifying it at once as first level property he have to worry about specifing it several times in device properties. Morover, if the same hardware device is used several times in different combinations (MULTI:HETERO,CPU where HETERO uses GPU,CPU) it became more and more complicated and unclear where desired property should be specified.
2. As far as I understood, it is not backward compatible. Because currently meta plugins silently accept properties which are supported by underlying hardware plugins. We can affect a lot of users. @ilya-lavrenov please correct me if I'm wrong
3. Related to latest point: how is it working right now? I suppose it should work as you described.